### PR TITLE
NOX-181 Refactor NoxDataBaseConfigurator.ConfigureEntity

### PR DESCRIPTION
**Pull request by msadikkoseaupa**

### CODE REVIEW

File: src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs
Comment (rochar) [inline src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs:1]: Move this types and extensions to Nox.Types.EntityFramework, this is only to be used by Db Configuration

File: src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+
+    public string Name { get; }
+    public string? Description { get; }
+    public bool IsReadonly { get; set; }
+    public bool IsLocalized { get; }
+    public bool IsRequired { get; }
+
+    public NoxType Type { get; }
+    public AreaTypeOptions? AreaTypeOptions { get; }
```
Comment (rochar) [inline src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs]: Just have a single property, an attribute only has a TypeOpton or does not have it at all
```suggestion
    public INoxTypeOptions? TypeOptions { get; }
```

File: src/Nox.Types.EntityFramework/Types/Text/TextDatabaseConfigurator.cs
Comment (rochar) [inline src/Nox.Types.EntityFramework/Types/Text/TextDatabaseConfigurator.cs:1]: we can extend `INoxTypeDatabaseConfigurator `to have a property to get the type options for that configuration, example for Text

`public INoxTypeOptions? GetTypeOptions(NoxSimpleTypeDefinition noxSimpleTypeDefinition);`

And for text for example:

`public INoxTypeOptions GetTypeOptions(NoxSimpleTypeDefinition noxSimpleTypeDefinition) => noxSimpleTypeDefinition.TextTypeOptions ?? new TextTypeOptions();
`

this way in DatabaseConfigurator you can do something like this when creating the AttributeConfiguration

`databaseConfiguration.ConfigureEntityProperty(codeGeneratorState, key.ToAttributeConfiguration(databaseConfiguration.GetTypeOptions(key)), entity, true, modelBuilder, builder)`

File: src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
+namespace Nox.Types;
+
+/// <summary>
+/// Represents the configuration for an attribute, encapsulating various settings and options.
+/// </summary>
+public class AttributeConfiguration
```
Comment (rochar) [inline src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs]:
```suggestion
public class NoxTypeDatabaseConfiguration
```

**Review by rochar — CHANGES_REQUESTED** [review]

File: src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs
Comment (msadikkoseaupa) [reply src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs:1]: File has been moved to the related project and renamed as desired

File: src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs (hunk 1 above)
Comment (msadikkoseaupa) [reply src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs]: Desired change has been applied

File: src/Nox.Types.EntityFramework/Types/Text/TextDatabaseConfigurator.cs
Comment (msadikkoseaupa) [reply src/Nox.Types.EntityFramework/Types/Text/TextDatabaseConfigurator.cs:1]: Implementation has been improved

File: src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs (hunk 2 above)
Comment (msadikkoseaupa) [reply src/Nox.Types.Abstractions/TypeDefinitions/AttributeConfiguration.cs]: File name has been updated

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**